public/app.js

```diff
@@ -1,77 +1,110 @@
-const FUNDING_SOURCES = [
-  paypal.FUNDING.PAYPAL,
-  paypal.FUNDING.PAYLATER,
-  paypal.FUNDING.VENMO,
-];
-
-function handleTransactionCases(details) {
-  // Three cases to handle:
-  //    (1) Recoverable INSTRUMENT_DECLINED -> call actions.restart()
-  //    (2) Other non-recoverable errors -> Show a failure message
-  //    (3) Successful transaction -> Show confirmation or thank you message
-
-  // This example reads a v2/checkout/orders capture response, propagated from the server
-  // You could use a different API or structure for your 'orderData'
-  const errorDetail = Array.isArray(details.details) && details.details[0];
-
-  if (errorDetail && errorDetail.issue === "INSTRUMENT_DECLINED") {
+function handleTransactionCases(orderData) {
+  const errorDetail = orderData?.details?.[0];
+  if (errorDetail?.issue === 'INSTRUMENT_DECLINED') {
+    // (1) Recoverable INSTRUMENT_DECLINED -> call actions.restart()
+    // recoverable state, per https://developer.paypal.com/docs/checkout/standard/customize/handle-funding-failures/
     return actions.restart();
-    // https://developer.paypal.com/docs/checkout/integration-features/funding-failure/
+  } else if (errorDetail) {
+    // (2) Other non-recoverable errors -> Show a failure message
+    throw new Error(
+      `${errorDetail.description} (${orderData.debug_id})`,
+    );
+  } else if (!orderData.purchase_units) {
```

Comment:
Not sure if we need the `if else's` with the early returns and thrown Errors. Just a thought.